src/bytes.rs

```diff
@@ -63,6 +79,16 @@ impl ByteBuf {
             bytes: Vec::with_capacity(cap)
         }
     }
+
+    pub fn as_vec(self) -> Vec<u8> {
+        self.bytes
+    }
+}
```

Comment:
> Then the From impl is generated in the standard library if I read this correctly.
Yes, the offending implementation is 
```rust
impl<T> From<T> for T
```
http://doc.rust-lang.org/src/core/convert.rs.html#152